src/tests/ftest/util/storage_utils.py

```diff
@@ -527,6 +536,99 @@ def get_controller_mapping(self):
 
         return controllers
 
+    def get_mounted_addresses(self):
+        """Get a list of addresses of devices which are mounted on one or more hosts.
+
+        Returns:
+            list: a list of addresses of devices which are mounted on one or more hosts
+        """
+        mounted_addresses = set()
+        self._log.debug('Detecting mounted addresses on %s', self._hosts)
+        for device, hosts in self._get_mounted_devices().items():
+            mounted_addresses.update(self._get_addresses(hosts, device))
+        return list(mounted_addresses)
+
+    def _get_mounted_devices(self):
+        """Get a dictionary of list of mounted device names per host.
+
+        Returns:
+            dict: a dictionary of mounted device name keys on host NodeSet values
+        """
+        def get_mounted_parent(block_device):
+            """Get the mounted parent kernel device name for the lsblk json block device entry.
+
+            Args:
+                block_device (dict): lsblk json block device entry
+
+            Returns:
+                str: mounted parent kernel device name or None
+            """
+            if block_device and 'children' in block_device:
+                for child in block_device['children']:
+                    if get_mounted_parent(child):
+                        return block_device['name']
+            if block_device and 'mountpoint' in block_device and block_device['mountpoint']:
+                return block_device['name']
+            return None
+
+        mounted_devices = {}
+        self._log.debug('  Detecting mounted device names on %s', self._hosts)
+        result = run_remote(self._log, self._hosts, 'lsblk --output NAME,MOUNTPOINT --json')
+        for data in result.output:
+            if not data.passed:
+                self._log.debug('  - Error detecting mounted devices on %s', data.hosts)
+                continue
+            try:
+                lsblk_data = yaml.safe_load('\n'.join(data.stdout))
+            except yaml.YAMLError as error:
+                self._log.debug(
+                    '  - Error processing mounted device information on %s: %s', data.hosts, error)
+                continue
```

Comment:
Should this split stdout/stderr just in case some weird debugging comes in the stderr?

Reply:
Yes, thanks.

Reply:
Also interesting that `yaml.safe_load('\n'.join(data.stdout))` works on json! Is there a specific reason to use `yaml.safe_load` instead of `json.loads`?